src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -560,6 +560,14 @@ private BoundExpression BindSimpleBinaryOperator(BinaryExpressionSyntax node, Di
 
             if (hasErrors)
             {
+                if (GetTupleCardinality(left).HasValue &&
+                    GetTupleCardinality(right).HasValue &&
```

Comment:
>GetTupleCardinality(right).HasValue [](start = 20, length = 35)

What if this is not a tuple, but there is an implicit user-defined conversion to the tuple type on the left?

Reply:
From LDM discussion, conversions *to* tuples and deconstructions are purposefully not contributing to tuple equality scenarios.

---
In reply to: [170490878](https://github.com/dotnet/roslyn/pull/23356#discussion_r170490878) [](ancestors = 170490878)